app/admin/users/page.js

```diff
@@ -0,0 +1,426 @@
+"use client"
+import React, { useEffect, useState } from 'react';
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  Paper,
+  Typography,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  TextField,
+  MenuItem,
+  Checkbox,
+} from '@mui/material';
+import StyledPaper from "../../components/styledComponents/StyledPaper";
+import { BlueButton, GreenButton, RedButton } from "../../components/styledComponents/StyledButton";
+import DesignTitel from "../../components/styledComponents/DesignTitel";
+
+const Page = () => {
+  const [users, setUsers] = useState([]);
+  const [filteredUsers, setFilteredUsers] = useState([]);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [open, setOpen] = useState(false);
+  const [userActionOpen, setUserActionOpen] = useState(false);
+  const [userDetailsOpen, setUserDetailsOpen] = useState(false);
+  const [editUserOpen, setEditUserOpen] = useState(false);
+  const [selectedUser, setSelectedUser] = useState(null);
+  const [selectedUserIds, setSelectedUserIds] = useState([]);
+  const [formData, setFormData] = useState({
+    email: '',
+    username: '',
+    first_name: '',
+    last_name: '',
+    telephone: '',
+    address: '',
+    group_id: '',
+  });
+
+  // Benutzer abrufen
```

Comment:
Hier hatte ich eine fertige Funktion, die auch auth für euch übernimmt. in der page.jsx ist die noch zu finden. Die Funktion erlaubt außerdem die Methode festzulegen, solte also auch CRUD funktionieren